# Description: use lsb informations to correctly display the distribution
# Upstream: http://bugzilla.gnome.org/show_bug.cgi?id=589549
# Ubuntu: https://bugs.launchpad.net/bugs/396805
--- ./gdm-2.27.4/gui/simple-greeter/gdm-greeter-login-window.c.gnome	2009-07-23 23:12:50.000000000 +0200
+++ ./gdm-2.27.4/gui/simple-greeter/gdm-greeter-login-window.c	2009-07-23 23:24:26.000000000 +0200
@@ -1265,10 +1265,16 @@
 static char *
 get_system_version (void)
 {
-        char *version;
+        char *version, *output;
         int i;
 
         version = NULL;
+        output = NULL;                
+
+        if (g_spawn_command_line_sync ("lsb_release -sd", &output, NULL, NULL, NULL)) {
+                version = g_strchomp (output);
+                return version;
+        }    
 
         for (i = 0; known_etc_info_files [i]; i++) {
                 char *path1;
@@ -1289,8 +1295,6 @@
         }
 
         if (version == NULL) {
-                char *output;
-                output = NULL;
                 if (g_spawn_command_line_sync ("uname -sr", &output, NULL, NULL, NULL)) {
                         version = g_strchomp (output);
                 }
